### PR TITLE
Refs #27808 -- Added test for saving nested ArrayField with nullable base field.

### PR DESCRIPTION
**Pull request by hramezani**
Test for [Ticket 27808](https://code.djangoproject.com/ticket/27808)

### CODE REVIEW

**Comment by pope1ni** [comment]
Do we want to pin `psycopg2-binary>=2.7.5` given that this was an issue fixed in Psycopg2?

File: tests/postgres_tests/models.py [hunk 1]
```diff
@@ -62,6 +62,10 @@ class NestedIntegerArrayModel(PostgreSQLModel):
     field = ArrayField(ArrayField(models.IntegerField()))
 
 
+class NestedNullableIntegerArrayModel(PostgreSQLModel):
+    field = ArrayField(ArrayField(models.IntegerField(null=True)))
```
Comment (felixxm) [inline tests/postgres_tests/models.py]: I don't think that we need a separate model, I would add 
```
field_nullable = ArrayField(ArrayField(models.IntegerField(null=True)))
```
to the `NestedIntegerArrayModel`.

File: tests/postgres_tests/test_array.py [hunk 2]
```diff
@@ -140,6 +141,15 @@ def test_model_set_on_base_field(self):
...
         self.assertEqual(field.base_field.model, IntegerArrayModel)
 
+    def test_nested_nullable_base_field(self):
+        instance = NestedNullableIntegerArrayModel.objects.create(
+            field=[[None, None], [None, None]]
+        )
+        self.assertSequenceEqual(
+            NestedNullableIntegerArrayModel.objects.all(),
+            [instance]
+        )
```
Comment (felixxm) [inline tests/postgres_tests/test_array.py]: We should take into account the `PSYCOPG2_VERSION` version, e.g.
```python
@unittest.skipIf(PSYCOPG2_VERSION < (2, 7, 5), 'https://github.com/psycopg/psycopg2/issues/325')
```

File: tests/postgres_tests/test_array.py [hunk 3]
```diff
@@ -140,6 +141,15 @@ def test_model_set_on_base_field(self):
         self.assertEqual(field.model, IntegerArrayModel)
         self.assertEqual(field.base_field.model, IntegerArrayModel)
 
+    def test_nested_nullable_base_field(self):
+        instance = NestedNullableIntegerArrayModel.objects.create(
+            field=[[None, None], [None, None]]
+        )
+        self.assertSequenceEqual(
+            NestedNullableIntegerArrayModel.objects.all(),
```
Comment (felixxm) [inline tests/postgres_tests/test_array.py]: Running query is not necessary, I think we can check field value instead, e.g.
```python
self.assertEqual(instance.field, [[None, None], [None, None]])
```

**Comment by hramezani** [comment]
@felixxm, @pope1ni Thanks for the review.
Don't know `PSYCOPG2_VERSION` checking is good enough or not.

**Comment by felixxm** [comment]
@hramezani Thanks :+1: I pushed minor edits.

**Pull request merged**